internal/server/server_test.go

```diff
@@ -98,3 +98,91 @@ func TestSOMASServerFactoryInitialisesClients(t *testing.T) {
 		}
 	}
 }
+
+func lstHasUniqueClientIDs(lst []shared.ClientID) bool {
+	// set to contain what we've seen so far
+	var s map[shared.ClientID]interface{}
+	for _, id := range lst {
+		if _, ok := s[id]; ok {
+			return false
+		}
+		s[id] = nil
+	}
+	return true
+}
+
+func TestGetNRandValuesFromLstUniqueWherePossible(t *testing.T) {
+	iterations := 10
+	cases := []struct {
+		name      string
+		input     []shared.ClientID
+		unique    bool
+		retLength int
+	}{
+		{
+			name:      "RandomAssign 2 entries",
+			input:     []shared.ClientID{shared.ClientID(0), shared.ClientID(1)},
+			unique:    false,
```

Comment:
nitpick: you dont really need `unique` here do you :) (unique is just len(input) < retLength)